### PR TITLE
feat:hyper theme project action support function

### PR DESCRIPTION
**Pull request by zhaogangang**
i want to custom action in project, like    
``` lua
            project = {
                action = function(path)
                    vim.fn.chdir(path)
                    require('telescope').extensions.frecency.frecency({ workspace = 'CWD' })
                end
            }
```
i think frecency more useful than find_files

### CODE REVIEW

**Comment by glepnir** [comment]
you also need dump it in cache config

**Comment by zhaogangang** [comment]
review it again,thanks

**Comment by glepnir** [comment]
i mean you need dump function to bytecode. ohterwise it will cause an error when read config from cache by using `Dashboard` command.

**Comment by zhaogangang** [comment]
i am sorry, i can't understand, i am rookie for write plugin, could you tell me how to do this

**Comment by glepnir** [comment]
`init.lua` cache_config function.

**Comment by zhaogangang** [comment]
it's like here? i test in my machine, it's work ok
``` lua
  if self.opts.config.project and type(self.opts.config.project.action) == 'function' then
    ---@diagnostic disable-next-line: param-type-mismatch
    local dump = assert(string.dump(self.opts.config.project.action))
    self.opts.config.project.action = dump
  end
```

**Comment by zhaogangang** [comment]
i read whole project, then i understand you said, thank you.

File: lua/dashboard/theme/hyper.lua [hunk 1]
```diff
@@ -101,7 +101,7 @@ local function project_list(config, callback)
     limit = 8,
     icon = ' ',
     icon_hl = 'DashboardRecentProjectIcon',
-    action = 'Telescope find_files cwd=',
+    action = function(path) vim.cmd("Telescope find_files cwd=" .. path) end,
```
Comment (glepnir) [inline lua/dashboard/theme/hyper.lua:104]: why changed here?

Comment (zhaogangang) [reply lua/dashboard/theme/hyper.lua:104]: this code can restored, it doesn't matter. i just write a sample for support function

Comment (glepnir) [reply lua/dashboard/theme/hyper.lua:104]: no need changed here

Comment (zhaogangang) [reply lua/dashboard/theme/hyper.lua:104]: ok, i restore it

File: README.md [hunk 2]
```diff
@@ -91,8 +91,14 @@ config = {
...
   packages = { enable = true }, -- show how many plugins neovim loaded
-  -- limit how many projects list, action when you press key or enter it will run this action.
-  project = { limit = 8, icon = 'your icon', label = '', action = 'Telescope find_files cwd=' },
+  -- limit how many projects list, action when you press key or enter it will run this action. 
+  -- @path param in action func is this project's path.
+  project = {
+      limit = 8,
+      icon = 'your icon',
+      label = '',
+      action = function(path) vim.cmd("Telescope find_files cwd=" .. path) end,
```
Comment (glepnir) [inline README.md:100]: need introduce this action type not write a function in there

**Pull request merged**